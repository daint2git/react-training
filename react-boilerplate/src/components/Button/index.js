import React from 'react'
import PropTypes from 'prop-types'
import { Button as BaseButton } from 'reactstrap'

import cssModuleNameTag from 'utils/cssModuleNameTag'

import styles from './styles.scss'

const cssModules = cssModuleNameTag(styles)

const Button = ({ className, children, ...other }) => (
  <BaseButton className={cssModules`root ${className}`} {...other}>
    {children}
  </BaseButton>
)

Button.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
}

export default Button
