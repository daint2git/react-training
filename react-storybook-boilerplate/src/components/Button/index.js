import React from 'react'
import styles from './index.scss'

const Button = props => <button className={styles.button} {...props} />

export default Button
