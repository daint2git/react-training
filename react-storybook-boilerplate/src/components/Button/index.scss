.button {
  font-size: 13px;
  color: white;
  background-color: #3498db;
  transition: all 200ms linear;
  cursor: pointer;

  &:hover {
    background-color: #e74c3c;
  }
}
