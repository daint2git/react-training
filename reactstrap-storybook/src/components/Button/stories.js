import React from 'react'
import { storiesOf } from '@storybook/react'
import { Button } from 'reactstrap'
import styles from './stories.scss'

const Wrapper = props => <div className={styles.wrapper} {...props} />

storiesOf('Button', module)
  .add('default', () => <Button>Hello Button</Button>)
  .add('Color', () => (
    <Wrapper>
      <Button color="primary">primary</Button>
      <Button color="secondary">secondary</Button>
      <Button color="success">success</Button>
      <Button color="info">info</Button>
      <Button color="warning">warning</Button>
      <Button color="danger">danger</Button>
      <Button color="link">link</Button>
    </Wrapper>
  ))
  .add('outline', () => (
    <Wrapper>
      <Button outline color="primary">
        primary
      </Button>
      <Button outline color="secondary">
        secondary
      </Button>
      <Button outline color="success">
        success
      </Button>
      <Button outline color="info">
        info
      </Button>
      <Button outline color="warning">
        warning
      </Button>
      <Button outline color="danger">
        danger
      </Button>
    </Wrapper>
  ))
  .add('size', () => (
    <Wrapper>
      <Button color="primary" size="lg">
        Large Button
      </Button>
      <Button color="secondary" size="lg">
        Large Button
      </Button>
      <Button color="primary" size="sm">
        Small Button
      </Button>
      <Button color="secondary" size="sm">
        Small Button
      </Button>
      <Button color="primary" size="lg" block>
        Block level button
      </Button>
      <Button color="secondary" size="lg" block>
        Block level button
      </Button>
    </Wrapper>
  ))
  .add('Active State', () => (
    <Wrapper>
      <Button color="primary" size="lg" active>
        Primary link
      </Button>
      <Button color="secondary" size="lg" active>
        Link
      </Button>
    </Wrapper>
  ))
  .add('Disabled State', () => (
    <Wrapper>
      <Button color="primary" size="lg" disabled>
        Primary button
      </Button>
      <Button color="secondary" size="lg" disabled>
        Button
      </Button>
    </Wrapper>
  ))
