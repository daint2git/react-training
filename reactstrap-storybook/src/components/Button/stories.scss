.wrapper {
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  padding-left: 20px;

  > button {
    margin-top: 10px;
  }
}
