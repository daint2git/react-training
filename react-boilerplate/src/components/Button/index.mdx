---
name: Button
---

import { Playground, Props } from 'docz'
import 'bootstrap/dist/css/bootstrap.css'
import 'sass/app.scss'
import ReactLogo from 'svgs/react-logo.svg'
import reduxLogo from 'svgs/redux-logo.svg'
import Button from './index'

# Button

<Props of={Button} />

## Basic usage

<Playground>
  <Button>Click</Button>
  <Button color="primary">Click</Button>


  <img src={reduxLogo} alt="redux-logo" width={20} height={20} />


</Playground>