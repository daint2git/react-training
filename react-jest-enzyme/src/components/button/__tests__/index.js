import 'jest-styled-components'
import Button from '../'

describe('Button', () => {
  test('converts basic button using shallow', () => {
    const shallowed = shallow(
      <Button className="class">
        <strong>Hello!</strong>
      </Button>,
    )

    expect(shallowed).toMatchSnapshot()
  })

  test('converts basic button using mount', () => {
    const mounted = mount(
      <Button className="class">
        <strong>Hello!</strong>
      </Button>,
    )

    expect(mounted).toMatchSnapshot()
    mounted.unmount()
  })

  test('converts basic button using render', () => {
    const rendered = render(
      <Button className="class">
        <strong>Hello!</strong>
      </Button>,
    )

    expect(rendered).toMatchSnapshot()
  })

  test('converts basic button using react-test-renderer', () => {
    const tree = renderer
      .create(
        <Button className="class">
          <strong>Hello!</strong>
        </Button>,
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
