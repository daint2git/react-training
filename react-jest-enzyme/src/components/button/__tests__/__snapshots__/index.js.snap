// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button converts basic button using mount 1`] = `
.c0 {
  color: white;
  background-color: black;
}

<styled.button
  className="class"
  type="button"
>
  <button
    className="class c0"
    type="button"
  >
    <strong>
      Hello!
    </strong>
  </button>
</styled.button>
`;

exports[`Button converts basic button using react-test-renderer 1`] = `
.c0 {
  color: white;
  background-color: black;
}

<button
  className="class c0"
  type="button"
>
  <strong>
    Hello!
  </strong>
</button>
`;

exports[`Button converts basic button using render 1`] = `
.c0 {
  color: white;
  background-color: black;
}

<button
  class="class c0"
  type="button"
>
  <strong>
    Hello!
  </strong>
</button>
`;

exports[`Button converts basic button using shallow 1`] = `
<styled.button
  className="class"
  type="button"
>
  <strong>
    Hello!
  </strong>
</styled.button>
`;
